sys/include/net/sock/udp.h

```diff
@@ -318,7 +318,8 @@ typedef struct sock_udp sock_udp_t;
  *
  * @return  0 on success.
  * @return  -EADDRINUSE, if `local != NULL` and @p local is already used
- *         elsewhere
+ *          elsewhere or if `local->port == 0` but the pool of ephemeral ports
```

Comment:
> but the pool of ephemeral ports is depleted.

I would not expect `-EADDRINUSE` for the above condition.

Rather use `-ENOBUFS` or `-ENOMEM` like specified [here](http://pubs.opengroup.org/onlinepubs/007904875/functions/sendto.html):

>[ENOBUFS]
    Insufficient resources were available in the system to perform the operation.
[ENOMEM]
    Insufficient memory was available to fulfill the request.

Reply:
In any case: this is not a memory problem. This is just a problem of a range of numbers not being available anymore. So I think both `ENOBUFS` and `ENOMEM` are sufficient. I wasn't able to find out how Linux syscalls are reporting this error, or if they even consider it.

Reply:
did you mean **in**sufficient?

Nevertheless, I had a quick look at linux. Looks like the case is not clearly handled there and oftentimes they also just return `-EADDRINUSE` or `-EAGAIN`. Let's keep it this way then.